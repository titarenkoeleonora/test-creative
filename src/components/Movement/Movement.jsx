import { useState, useRef } from "react";

const Movement = () => {
  const arrayInputElement = useRef(null);
  const countInputElement = useRef(null);
  const [array, setArray] = useState([]);
  const [movementCount, setMovementCount] = useState(0);

  const getArrayInputValues = () => {
    setArray(arrayInputElement.current.value.split(""));
  };

  const getCountInputValue = () => {
    setMovementCount(countInputElement.current.value);
  };

  const moveArray = (evt) => {
    evt.preventDefault();

    console.log("Входной массив", arrayInputElement.current.value.split(""));
    console.log("Кол-во шагов", countInputElement.current.value);
    console.log(
      "Выходной массив",
      array.concat(array.splice(0, array.length - movementCount))
    );
  };

  return (
    <section className="movement">
      <div className="container">
        <div className="movement__wrapper">
          <h2 className="movement__title">
            Функция сдвига элементов массива вправо на N шагов
          </h2>
          <form className="movement__form" method="post">
            <label className="movement__label" id="movement-array">
              Введите числа
            </label>
            <input
              className="movement__input input-text"
              type="number"
              htmlFor="movement-array"
              ref={arrayInputElement}
              onInput={getArrayInputValues}
              required
            />
            <label className="movement__label" id="movement-count">
              Введите кол-во шагов для передвижения массива
            </label>
            <input
              className="movement__input  input-text"
              type="number"
              htmlFor="movement-count"
              ref={countInputElement}
              onInput={getCountInputValue}
              required
            />
            <div className="movement__buttons-wrapper">
              <button
                className="movement__submit-button button"
                type="submit"
                onClick={moveArray}
              >
                Показать результат
              </button>
              <button className="movement__reset-button button" type="reset">
                Очистить
              </button>
            </div>
          </form>

          <p className="movement__result-text">
            Для просмотра результата откройте консоль (F12)
          </p>
        </div>
      </div>
    </section>
  );
};

export default Movement;
