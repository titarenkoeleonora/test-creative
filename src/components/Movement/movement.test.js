import React from "react";
import renderer from "react-test-renderer";
import Movement from "./Movement";

it('Render Movement component', () => {
  const tree = renderer
    .create(
      <Movement />
    );

  expect(tree).toMatchSnapshot();
});