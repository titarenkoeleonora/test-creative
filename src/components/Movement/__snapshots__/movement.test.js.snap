// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movement component 1`] = `
<section
  className="movement"
>
  <div
    className="container"
  >
    <div
      className="movement__wrapper"
    >
      <h2
        className="movement__title"
      >
        Функция сдвига элементов массива вправо на N шагов
      </h2>
      <form
        className="movement__form"
        method="post"
      >
        <label
          className="movement__label"
          id="movement-array"
        >
          Введите числа или буквы
        </label>
        <input
          className="movement__input input-text"
          htmlFor="movement-array"
          onInput={[Function]}
          pattern="/\\\\d+/g"
          required={true}
          type="text"
        />
        <label
          className="movement__label"
          id="movement-count"
        >
          Введите кол-во шагов для передвижения массива
        </label>
        <input
          className="movement__input  input-text"
          htmlFor="movement-count"
          onInput={[Function]}
          required={true}
          type="number"
        />
        <div
          className="movement__buttons-wrapper"
        >
          <button
            aria-label="Показать результат"
            className="movement__submit-button button"
            onClick={[Function]}
            type="submit"
          >
            Показать результат
          </button>
          <button
            aria-label="Очистить форму"
            className="movement__reset-button button"
            onClick={[Function]}
            type="reset"
          >
            Очистить
          </button>
        </div>
      </form>
      <p
        className="movement__result-text"
      >
        Для просмотра результата откройте консоль (F12)
      </p>
    </div>
  </div>
</section>
`;
